apis/outlook/mail/code/pkg/graph/mail.go

```diff
@@ -36,32 +36,60 @@ func GetMessageDetails(ctx context.Context, client *msgraphsdkgo.GraphServiceCli
 }
 
 func SearchMessages(ctx context.Context, client *msgraphsdkgo.GraphServiceClient, query, folderID string) ([]models.Messageable, error) {
+	// We search specifically by subject using $filter and then using $search.
+	// The search results from $search are often not all that great for whatever reason. The subject line is probably more important.
+	// So we combine the results with the subject ones first, and then dedupe and return.
+
 	var (
-		result models.MessageCollectionResponseable
-		err    error
+		subjectResult models.MessageCollectionResponseable
+		subjectErr    error
+		result        models.MessageCollectionResponseable
+		err           error
 	)
 
 	if folderID != "" {
+		subjectResult, subjectErr = client.Me().MailFolders().ByMailFolderId(folderID).Messages().Get(ctx, &users.ItemMailFoldersItemMessagesRequestBuilderGetRequestConfiguration{
+			QueryParameters: &users.ItemMailFoldersItemMessagesRequestBuilderGetQueryParameters{
+				Filter: util.Ptr(fmt.Sprintf("contains(subject, '%s')", query)),
+				Top:    util.Ptr(int32(10)),
+			},
+		})
+
 		result, err = client.Me().MailFolders().ByMailFolderId(folderID).Messages().Get(ctx, &users.ItemMailFoldersItemMessagesRequestBuilderGetRequestConfiguration{
 			QueryParameters: &users.ItemMailFoldersItemMessagesRequestBuilderGetQueryParameters{
 				Search: &query,
-				Top:    util.Ptr(int32(50)),
+				Top:    util.Ptr(int32(10)),
 			},
 		})
 	} else {
+		subjectResult, subjectErr = client.Me().Messages().Get(ctx, &users.ItemMessagesRequestBuilderGetRequestConfiguration{
+			QueryParameters: &users.ItemMessagesRequestBuilderGetQueryParameters{
+				Filter: util.Ptr(fmt.Sprintf("contains(subject, '%s')", query)),
+				Top:    util.Ptr(int32(10)),
+			},
+		})
+
 		result, err = client.Me().Messages().Get(ctx, &users.ItemMessagesRequestBuilderGetRequestConfiguration{
 			QueryParameters: &users.ItemMessagesRequestBuilderGetQueryParameters{
 				Search: &query,
-				Top:    util.Ptr(int32(50)),
+				Top:    util.Ptr(int32(10)),
 			},
 		})
 	}
 
+	if subjectErr != nil {
+		return nil, fmt.Errorf("failed to search messages by subject: %w", subjectErr)
+	}
 	if err != nil {
 		return nil, fmt.Errorf("failed to search messages: %w", err)
 	}
 
-	return result.GetValue(), nil
+	var fullResults []models.Messageable
+	fullResults = append(fullResults, subjectResult.GetValue()...)
+	fullResults = append(fullResults, result.GetValue()...)
```

Comment:
This feels dumb but I guess there's a chance that `subjectResult` or `result` could be nil? Didn't wanna try to append to those.